### PR TITLE
Update spectral library with higher score

### PR DESCRIPTION
**Pull request by YulingDai**

### CODE REVIEW

**Review by trishorts — CHANGES_REQUESTED** [review]

File: GUI/GUI.csproj [hunk 1]
```diff
@@ -133,4 +133,10 @@
       <CopyToOutputDirectory>Always</CopyToOutputDirectory>
     </None>
   </ItemGroup>
+
```
[the remark this answers is not in the document]
Comment (Yuling1996) [reply GUI/GUI.csproj:136]: Deleted in the new pull request

File: TaskLayer/SearchTask/PostSearchAnalysisTask.cs [hunk 2]
```diff
@@ -585,6 +591,48 @@ private void WritePsmResults()
             }
         }
 
+        private void UpdateSpectralLibrary()
+        {
+            var FilteredPsmList = Parameters.AllPsms
+               .Where(p => p.FdrInfo.PEP_QValue <= 0.01 && p.FdrInfo.QValueNotch <= CommonParameters.QValueOutputFilter).ToList();
```
[the remark this answers is not in the document]
Comment (Yuling1996) [reply TaskLayer/SearchTask/PostSearchAnalysisTask.cs:597]: Did some changes and added more comments

File: GUI/TaskWindows/SearchTaskWindow.xaml [hunk 3]
```diff
@@ -436,9 +436,10 @@
                         </StackPanel>
                     </GroupBox>
 
-                    <!-- Output Options -->
+                   <!-- Output Options -->
                     <GroupBox Header="Output Options" DockPanel.Dock="Top" ScrollViewer.CanContentScroll="True">
 
+                        <!--<Expander x:Name="OutputExpander">-->
```
[the remark this answers is not in the document]
Comment (YulingDai) [reply GUI/TaskWindows/SearchTaskWindow.xaml]: deleted

**Review by nbollis — APPROVED** [review]

File: TaskLayer/SearchTask/PostSearchAnalysisTask.cs [hunk 4]
```diff
@@ -585,6 +591,54 @@ private void WritePsmResults()
             }
         }
 
+        private void UpdateSpectralLibrary()
```
Comment (Alexander-Sol) [inline TaskLayer/SearchTask/PostSearchAnalysisTask.cs:594]: May want to consider taking these functions out of the PostSearchAnalysisTask and moving them into one of the SpectralLibrary classes. Not mandatory, but might make things more organized.

**Review by Alexander-Sol — COMMENTED** [review]
The spectral library object maintains a persistent connection to the spectral library file.

Attempting to deleted the updatedLib file at the end of TestLibraryUpdate throws an error, saying that the file is still in use. This is true even if you call updatedLib.CloseConnections. Should this be fixed?

File: TaskLayer/SearchTask/PostSearchAnalysisTask.cs (hunk 4 above)
Comment (YulingDai) [reply TaskLayer/SearchTask/PostSearchAnalysisTask.cs:594]: Thanks for your advice! I put the library update function here as the function of spectral library generation is here. They are the same kind of functions.

**Review by Alexander-Sol — APPROVED** [review]

**Review by trishorts — APPROVED** [review]

**Pull request merged**